azure/services/aso/service.go

```diff
@@ -62,18 +66,17 @@ func (s *Service[T, S]) Reconcile(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, s.Scope.DefaultedAzureServiceReconcileTimeout())
 	defer cancel()
 
-	if len(s.Specs) == 0 {
```

Comment:
What explains this deletion of returning early in case of an empty s.Specs array?